receiver/nsxtreceiver/scraper.go

```diff
@@ -0,0 +1,253 @@
+
+		r = append(r, nodeInfo)
+	}
+
+	wg.Wait()
+
+	return r, errs.Combine()
+}
+
+func (s *scraper) retrieveInterfaces(
+	ctx context.Context,
+	nodeProps dm.NodeProperties,
+	nodeInfo *nodeInfo,
+	nodeClass nodeClass,
+	wg *sync.WaitGroup,
+	errs *scrapererror.ScrapeErrors,
+) {
+	defer wg.Done()
+	interfaces, err := s.client.Interfaces(ctx, nodeProps.ID, nodeClass)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+	nodeInfo.interfaces = []interfaceInformation{}
+	for _, i := range interfaces {
+		interfaceInfo := interfaceInformation{
+			iFace: i,
+		}
+		stats, err := s.client.InterfaceStatus(ctx, nodeProps.ID, i.InterfaceId, nodeClass)
+		if err != nil {
+			errs.AddPartial(1, err)
+		}
+		interfaceInfo.stats = stats
+		nodeInfo.interfaces = append(nodeInfo.interfaces, interfaceInfo)
+	}
+}
+
+func (s *scraper) retrieveNodeStats(
+	ctx context.Context,
+	nodeProps dm.NodeProperties,
+	nodeInfo *nodeInfo,
+	nodeClass nodeClass,
+	wg *sync.WaitGroup,
+	errs *scrapererror.ScrapeErrors,
+) {
+	defer wg.Done()
+	ns, err := s.client.NodeStatus(ctx, nodeProps.ID, nodeClass)
+	if err != nil {
+		errs.AddPartial(1, err)
+		return
+	}
+	nodeInfo.stats = ns
+}
+
+func (s *scraper) process(
+	nodes []*nodeInfo,
+	colTime pdata.Timestamp,
+) {
+	for _, n := range nodes {
+		for _, i := range n.interfaces {
+			s.recordNodeInterface(colTime, n.nodeProps, i)
+		}
+		s.recordNode(colTime, n)
+	}
+}
+
+func (s *scraper) recordNodeInterface(colTime pdata.Timestamp, nodeProps dm.NodeProperties, i interfaceInformation) {
+	s.mb.RecordNsxtNodeNetworkPacketCountDataPoint(colTime, i.stats.RxDropped, metadata.AttributeDirectionReceived, metadata.AttributePacketTypeDropped)
+	s.mb.RecordNsxtNodeNetworkPacketCountDataPoint(colTime, i.stats.RxErrors, metadata.AttributeDirectionReceived, metadata.AttributePacketTypeErrored)
+	successRxPackets := i.stats.RxPackets - i.stats.RxDropped - i.stats.RxErrors
+	s.mb.RecordNsxtNodeNetworkPacketCountDataPoint(colTime, successRxPackets, metadata.AttributeDirectionReceived, metadata.AttributePacketTypeSuccess)
+
+	s.mb.RecordNsxtNodeNetworkPacketCountDataPoint(colTime, i.stats.TxDropped, metadata.AttributeDirectionTransmitted, metadata.AttributePacketTypeDropped)
+	s.mb.RecordNsxtNodeNetworkPacketCountDataPoint(colTime, i.stats.TxErrors, metadata.AttributeDirectionTransmitted, metadata.AttributePacketTypeErrored)
+	successTxPackets := i.stats.TxPackets - i.stats.TxDropped - i.stats.TxErrors
+	s.mb.RecordNsxtNodeNetworkPacketCountDataPoint(colTime, successTxPackets, metadata.AttributeDirectionTransmitted, metadata.AttributePacketTypeSuccess)
+
+	s.mb.RecordNsxtNodeNetworkIoDataPoint(colTime, i.stats.RxBytes, metadata.AttributeDirectionReceived)
+	s.mb.RecordNsxtNodeNetworkIoDataPoint(colTime, i.stats.TxBytes, metadata.AttributeDirectionTransmitted)
+
+	s.mb.EmitForResource(
+		metadata.WithDeviceID(i.iFace.InterfaceId),
+		metadata.WithNsxtNodeName(nodeProps.Name),
```

Comment:
Do we also know the nodeType or node ID for network metrics?

Reply:
Good catch! just pushed a change that includes them